### PR TITLE
sdcv: 0.4.2 -> 0.5.2

### PR DESCRIPTION
**Pull request by pallavagarwal07**
###### Motivation for this change
The build was no longer working on latest fork of nixos-unstable.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/sdcv/default.nix [hunk 1]
```diff
@@ -1,7 +1,9 @@
-{ stdenv, fetchurl, pkgconfig, glib, gettext }:
+{ stdenv, fetchurl, pkgconfig, glib, gettext, fetchFromGitHub, cmake,
```
Comment (c0bw3b) [inline pkgs/applications/misc/sdcv/default.nix]: `fetchFromGitHub` should replace `fetchurl` since you are not using the latter anymore

File: pkgs/applications/misc/sdcv/default.nix [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+readline }:
 
 stdenv.mkDerivation rec {
-  name= "sdcv-0.4.2";
+  name = "sdcv-${version}";
+  version = "0.5.2";
 
   meta = {
     homepage = http://sdcv.sourceforge.net/;
```
Comment (c0bw3b) [inline pkgs/applications/misc/sdcv/default.nix]: This page now redirects to https://dushistov.github.io/sdcv/ so we should set `meta.homepage` to this new home.

File: pkgs/applications/misc/sdcv/default.nix [hunk 3]
```diff
@@ -11,25 +13,21 @@ stdenv.mkDerivation rec {
...
-              else [ ./sdcv.cpp.patch ] );
-
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ cmake pkgconfig readline ];
   buildInputs = [ glib gettext ];
 
-  preBuild = ''
-    sed -i 's/guint32 page_size/size_t page_size/' src/lib/lib.cpp
+  preInstall = ''
+    touch locale
```
Comment (c0bw3b) [inline pkgs/applications/misc/sdcv/default.nix]: Just curious: why is this necessary?

**Review by c0bw3b — CHANGES_REQUESTED** [review]
Thanks for your help refreshing an old package. :)

File: pkgs/applications/misc/sdcv/utils.hpp.patch [hunk 4]
```diff
@@ -1,10 +0,0 @@
---- a/src/utils.hpp
-+++ b/src/utils.hpp
-@@ -2,6 +2,7 @@
- #define _UTILS_HPP_
- 
- #include <string>
-+#include <cstdlib>
```
Comment (yegortimoshenko) [inline pkgs/applications/misc/sdcv/utils.hpp.patch]: I don't see this patch at upstream: https://github.com/Dushistov/sdcv/blob/v0.5.2/src/utils.hpp

File: pkgs/applications/misc/sdcv/default.nix (hunk 3 above)
Comment (yegortimoshenko) [reply pkgs/applications/misc/sdcv/default.nix]: Without this:

```
CMake Error at cmake_install.cmake:60 (file):
  file INSTALL cannot find
  "/tmp/nix-build-sdcv-0.5.2.drv-0/source/build/locale".
```

File: pkgs/applications/misc/sdcv/utils.hpp.patch (hunk 4 above)
Comment (orivej) [reply pkgs/applications/misc/sdcv/utils.hpp.patch]: Travis passes.

**Review by orivej — APPROVED** [review]
